geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/EnumeratedHistogram.scala

```diff
@@ -0,0 +1,65 @@
+/***********************************************************************
+  * Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+  * All rights reserved. This program and the accompanying materials
+  * are made available under the terms of the Apache License, Version 2.0 which
+  * accompanies this distribution and is available at
+  * http://www.opensource.org/licenses/apache2.0.php.
+  *************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.mutable
+import scala.util.parsing.json.JSONObject
+
+/**
+ * An EnumeratedHistogram is merely a HashMap mapping values to number of occurences
+ * .
+ * @param attrIndex attribute index for the attribute the histogram is being made for
+ * @param attrType class type as a string for serialization purposes
+ * @tparam T some type T (which is restricted by the stat parser upstream of EnumeratedHistogram instantiation)
+ */
+class EnumeratedHistogram[T](val attrIndex: Int,
+                             val attrType: String) extends Stat {
+  val frequencyMap: mutable.Map[T, Long] = new mutable.HashMap[T, Long]().withDefaultValue(0)
+
+  override def observe(sf: SimpleFeature): Unit = {
+    val sfval = sf.getAttribute(attrIndex)
+    if (sfval != null) {
+      sfval match {
+        case tval: T =>
+          frequencyMap(tval) += 1
```

Comment:
Maybe do ```frequencyMap(sfval.asInstanceOf[T]) += 1``` instead? Seems like that would be faster than a match, and since the match doesn't have any other cases, it is functionally equivalent.